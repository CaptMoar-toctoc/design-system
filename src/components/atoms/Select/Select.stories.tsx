import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Select from './Select';

export default {
	title: 'TocToc-DS/Atoms',
	component: Select,
	// argTypes: {
	// 	checked:{control: 'boolean'}
	// },
} as ComponentMeta<typeof Select>;

const Template: ComponentStory<typeof Select> = (args) => <Select {...args} />;

export const SelectTT = Template.bind({});


SelectTT.args = {
	label:"Mi select",
	name:"miSelect",
	disabled: false,
	defaultValue: "",
	error: false,
	handlerFn: () => { console.log(`click`) },
	options: [
		{ label: "La Florida", value: "1" },
		{ label: "Las Condes", value: "2" },
		{ label: "Peñalolén", value: "3" },
	]
};
SelectTT.decorators = [
	(Story) => (
		<div style={{ maxWidth: '30%' }}>
			<Story />
		</div>
	),
];
