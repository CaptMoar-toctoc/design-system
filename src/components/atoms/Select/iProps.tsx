import { ReactNode } from "react";

export interface iProps {
    //children?: ReactNode;
	label: string,
	name: string,
	value?: any,
	options: any,
    handlerFn?: any;
    disabled?: boolean,
	defaultValue?: any,
	error?:boolean,
	helperText?: ReactNode
}