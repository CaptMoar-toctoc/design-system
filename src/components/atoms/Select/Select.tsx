import React, { FC } from 'react';
import { iProps } from './iProps';
import { ThemeProvider, Select as SelectTT } from '@mui/material';
import { SelectChangeEvent } from '@mui/material/Select';
import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import FormHelperText from '@mui/material/FormHelperText';

import theme from '../../theme/theme';

const Select: FC<iProps> = ({
	label,
	name,
	value,
	handlerFn,
	disabled,
	options,
	defaultValue,
	error,
	helperText
}) => {


	const OptionsList = options.map(function(op:any){
		return <MenuItem key={`k-${op.value}`} value={op.value}>{op.label}</MenuItem>
	});

	return (
		<ThemeProvider theme={theme}>

			<FormControl fullWidth error={error}>
				<InputLabel id={`${name}-select-label`}>{label}</InputLabel>
				<SelectTT
					labelId={`${name}-select-label`}
					id={`${name}-select`}
					value={value}
					label={label}
					disabled={disabled}
					defaultValue={ defaultValue }
					autoWidth={false}
					displayEmpty={true}
				// onChange={handleChange}
				>
					{OptionsList}
				</SelectTT>
				<FormHelperText>Hola, soy texto</FormHelperText>
			</FormControl>

		</ThemeProvider>
	);
}

export default Select;