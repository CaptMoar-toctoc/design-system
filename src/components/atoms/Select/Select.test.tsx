import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';


import Select from "./Select";

describe('Input Testing Render', () => {
	//inicializacion de props componente
	//aca van todos los props requerido y OPCIONALES clea
	const props = {
		label: 'Holaaaa',
		name: 'NameSelect',
		defaultValue:"1",
		options: [
			{ label: "La Florida", value: "1" },
			{ label: "Las Condes", value: "2" },
			{ label: "Peñalolén", value: "3" }
		]
	}

	test("Render SnapShot", () => {
		expect(render(<Select {...props} />)).toMatchSnapshot();
	})


})

