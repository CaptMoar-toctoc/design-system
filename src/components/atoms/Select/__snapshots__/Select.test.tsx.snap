// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing Render Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1ytumr6-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          id="NameSelect-select-label"
        >
          Holaaaa
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-mupwgz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="NameSelect-select-label NameSelect-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-q1syya-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="NameSelect-select"
            role="button"
            tabindex="0"
          >
            La Florida
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="1"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Holaaaa
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-idv3ut-MuiFormHelperText-root"
        >
          Hola, soy texto
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1ytumr6-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="NameSelect-select-label"
      >
        Holaaaa
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-mupwgz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="NameSelect-select-label NameSelect-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-q1syya-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="NameSelect-select"
          role="button"
          tabindex="0"
        >
          La Florida
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Holaaaa
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-idv3ut-MuiFormHelperText-root"
      >
        Hola, soy texto
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
